src/parsec.rs

```diff
@@ -15,25 +15,59 @@ use crate::{
     utils::LogIdent,
 };
 use log::LogLevel;
+use maidsafe_utilities::serialisation;
 #[cfg(not(feature = "mock_parsec"))]
 use parsec as inner;
 use rand::Rng;
-use std::collections::{btree_map::Entry, BTreeMap};
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    fmt,
+};
 
 #[cfg(feature = "mock_parsec")]
 pub use crate::mock::parsec::{
-    init_mock, ConsensusMode, NetworkEvent, Observation, Proof, PublicId, SecretId,
+    init_mock, ConsensusMode, Error, NetworkEvent, Observation, Proof, PublicId, SecretId,
 };
 #[cfg(not(feature = "mock_parsec"))]
-pub use parsec::{ConsensusMode, NetworkEvent, Observation, Proof, PublicId, SecretId};
+pub use parsec::{ConsensusMode, Error, NetworkEvent, Observation, Proof, PublicId, SecretId};
 
 pub type Block = inner::Block<chain::NetworkEvent, id::PublicId>;
 pub type Parsec = inner::Parsec<chain::NetworkEvent, FullId>;
 pub type Request = inner::Request<chain::NetworkEvent, id::PublicId>;
 pub type Response = inner::Response<chain::NetworkEvent, id::PublicId>;
 
+// TODO: we'll set PARSEC_SIZE_LIMIT to 1 GB once it's used outside of mock_parsec
+//#[cfg(not(feature = "mock_parsec"))]
+//const PARSEC_SIZE_LIMIT: u64 = 1_000_000_000;
+// Mock parsec request/responses are much smaller, so we need a lower limit.
+// TODO: once it's used outside of tests, this should be changed to cfg(feature = "mock_parsec")
+#[cfg(all(test, feature = "mock_parsec"))]
+const PARSEC_SIZE_LIMIT: u64 = 100;
+
+// Keep track of size in case we need to prune.
+#[derive(Default, Debug, PartialEq, Eq)]
+struct ParsecSizeCounter(u64);
+
+impl ParsecSizeCounter {
+    fn increase_size(&mut self, size: u64) {
+        self.0 += size;
+    }
+
+    #[cfg(all(test, feature = "mock_parsec"))]
+    fn needs_pruning(&self) -> bool {
```

Comment:
The feature guards reminds me that whether This PR shall contain the part of `returning` the counter status for the later on `vote for prune` ?
That is, the return of the `ParsecSiceCounter::increase_size()` and `Parsec::count_size()` shall incidicates a `overflow` and trigering an empty vote function to be called?
As the titile of the PR indicates this shall be part of the `detection` ?

Reply:
Do you mean something like
```
    fn increase_size(&mut self, size: u64) -> bool {
        self.0 += size;
        self.needs_pruninig()
    }
```
?

Reply:
I'm not seeing what that would be coupled, can look at the follow up PR to see how it is used.

Reply:
Well I'm of the opinion that if these needs to tweaked depending on how they're consumed then that should happen in that PR

Reply:
What I mean is, looking at the follow up PR, which exists, we can see that the existing API is fine?

Reply:
i.e no need to return a bool?